### PR TITLE
fix: to large data fetch in virtualized table

### PR DESCRIPTION
**Pull request by cbt1**
Fixes an issue where the virt table would call engine with a negative qTop value when fetching data.

### CODE REVIEW

File: src/table/components/virtualized-table/hooks/use-infinite-scroll-data.ts [hunk 1]
```diff
@@ -47,8 +48,7 @@ const useInfiniteScrollData = (
   pageInfo: PageInfo
 ): UseInfiniteScrollData => {
   const [rowsInPage, setRowsInPage] = useState<Row[]>([]);
-
-  useEffect(() => setRowsInPage([]), [layout, pageInfo]);
+  useOnPropsChange(() => setRowsInPage([]), [layout, pageInfo]);
```
Comment (cbt1) [inline src/table/components/virtualized-table/hooks/use-infinite-scroll-data.ts:51]: By not using useEffect here we get 1 less render cycle. Come out of me trying to fix the original issue but ended up being valuable.

File: src/table/components/virtualized-table/TableContainer.tsx [hunk 2]
```diff
@@ -85,4 +85,9 @@ export default function TableContainer(props: TableContainerProps) {
       </FullSizeContainer>
     </div>
   );
-}
+};
+
+// Export non memoized version for testing purpose
+export { TableContainer };
+
+export default memo(TableContainer);
```
Comment (cbt1) [inline src/table/components/virtualized-table/TableContainer.tsx:93]: On the initial render the `Wrapper` component got an extra render due to the `keyboard` prop being updated. For which the was no reason to re-render so added the memo.

File: src/table/components/virtualized-table/TableContainer.tsx [hunk 3]
```diff
@@ -85,4 +85,9 @@ export default function TableContainer(props: TableContainerProps) {
       </FullSizeContainer>
     </div>
   );
-}
+};
+
+// Export non memoized version for testing purpose
+export { TableContainer };
```
Comment (haxxmaxx) [inline src/table/components/virtualized-table/TableContainer.tsx:91]: for the record it is possible to test when you export memo(someComponent) but the syntax is weird so I like this better actually

File: src/table/components/virtualized-table/TableContainer.tsx (hunk 2 above)
Comment (haxxmaxx) [reply src/table/components/virtualized-table/TableContainer.tsx:93]: nice that you are adding memo very intentionally

File: src/table/components/virtualized-table/Wrapper.tsx [hunk 4]
```diff
@@ -1,21 +1,26 @@
-import React, { useState } from 'react';
+import React, { useMemo, useState } from 'react';
 import { WrapperProps } from './types';
 import PaginationContent from '../PaginationContent';
 import { StyledTableWrapper } from '../../styles';
 import { PageInfo, TableData } from '../../../types';
 import FooterWrapper from '../FooterWrapper';
-import TableContainer from './TableContainer';
+import Table from './TableContainer';
```
Comment (haxxmaxx) [inline src/table/components/virtualized-table/Wrapper.tsx:7]: how come the rename? I think it is good to keep the name of the component and the file the same

File: src/table/components/virtualized-table/hooks/use-on-props-change.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { useState } from 'react';
+
+// Act as a wrapper for https://beta.reactjs.org/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes
+export default function useOnPropsChange<T>(callback: () => void, props: T[]) {
```
Comment (haxxmaxx) [inline src/table/components/virtualized-table/hooks/use-on-props-change.ts:4]: nice one

**Review by haxxmaxx — APPROVED** [review]
lgtm, some minor things

File: src/table/components/virtualized-table/Wrapper.tsx (hunk 4 above)
Comment (cbt1) [reply src/table/components/virtualized-table/Wrapper.tsx:7]: Good question. Got lint issues when the memoized TableContainer and TableContainer was both exported with the same name, but the memoized version as default. Did not look further into how to best solve it. But I do agree that I would prefer the same name.

**Pull request merged**